File: codility/cpp/codility_lesson9_1.cpp
// you can use includes, for example:
// #include <algorithm>
#include <algorithm>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(vector<int> &A) {
	// write your code in C++14 (g++ 6.2.0)
	if (A.empty() || A.size() == 1)
		return 0;

	int N = A.size();
	int minVal = A[0];
	int maxProfit = 0;

	for (int i = 1; i < N; i++)
	{
		maxProfit = max(maxProfit, A[i] - minVal);
		minVal = min(minVal, A[i]);
	}

	return maxProfit;
}

File: codility/python/codility_lesson9_1.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
	# write your code in Python 3.6
	ans = 0
	if len(A) != 0:
		minval = A[0]
		for a in A:
			if a - minval > ans:
				ans = a - minval
			if a < minval:
				minval = a
	return ans
